enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,539 @@
+expectation that persistent storage availability and scale concerns will be a
+better fit for running a full RDBMS.
+
+A suggestion has been made that in situations where there is a single process
+running the assisted service, locking can happen in-memory instead of in the
+database. Further analysis is required.
+
+#### baremetal-operator watching multiple namespaces?
+
+When utilizing baremetal-operator on a Hub cluster to boot the discovery ISO on
+hosts, should we be creating those BareMetalHost resources in a separate
+namespace from those that are associated with the Hub cluster itself?
+
+What work is involved in having BMO watch additional namespaces?
+
+Upstream, the metal3 project is already running baremetal-operator watching
+multiple namespaces, so there should not be software changes required.
+
+#### Centralized Machine Management
+
+OpenShift's multi-cluster management is moving toward a centralized Machine
+management pattern, as an addition to the current approach where Machines only
+exist in the same cluster as their associated Node. This proposal should be
+compatible with centralized Machine management, but it would be useful to play
+through those workflows in detail to be certain.
+
+#### Garbage Collection
+
+Agent resources are not useful after provisioning except possibly as a
+historical record. A plan should be in place for garbage collecting them.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+- What additional testing is necessary to support managed OpenShift service-based offerings?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+New APIs will start at alpha versions with all of the functionality described
+in this proposal. Releases will be made available to key stakeholders for
+evaluation and testing in order to gather feedback. APIs may change during this
+time, but disruption will be minimized so as to encourage and facilitate
+testing.
+
+Once sufficient feedback has been collected from users (including lessons
+learned from ongoing formal testing), and a set of changes have been agreed
+upon, beta versions of APIs will be implemented and released alongside a future
+OpenShift version.
+
+### Upgrade / Downgrade Strategy
+
+The assisted service, its new controllers, and its CRDs will be distributed
+with Hive as part of one operator installed by Operator Lifecycle Manager. They
+will continue to follow the upgrade approach that Hive already follows.
```

Comment:
I would clarify here that we expect this to be the ACM operator bundling them together. Today community OCM has a dependency on Hive's community OLM operator, and the downstream official ACM directly embeds it within their single operator.